tests/unittest_scoped_nodes.py

```diff
@@ -291,6 +291,51 @@ def test_stream_api(self) -> None:
             with open(path, "rb") as file_io:
                 self.assertEqual(stream.read(), file_io.read())
 
+    def test_with_singleline_docstring(self) -> None:
+        data = """
+            '''Hello World'''
+            foo = 1
+        """
+        module = builder.parse(data, __name__)
+        assert isinstance(module.doc_node, nodes.Const)
+        assert module.doc_node.lineno == 2
+        assert module.doc_node.col_offset == 0
+        if PY38_PLUS:
+            assert module.doc_node.end_lineno == 2
+            assert module.doc_node.end_col_offset == 17
+        else:
+            assert module.doc_node.end_lineno is None
+            assert module.doc_node.end_col_offset is None
+
+    def test_with_multiline_docstring(self) -> None:
+        data = """
+            '''Hello World
+
+            Also on this line.
+            '''
+            foo = 1
+        """
+        module = builder.parse(data, __name__)
+
+        assert isinstance(module.doc_node, nodes.Const)
+        if PY38_PLUS:
+            assert module.doc_node.lineno == 2
+            assert module.doc_node.col_offset == 0
+            assert module.doc_node.end_lineno == 5
+            assert module.doc_node.end_col_offset == 3
+        else:
+            assert module.doc_node.lineno == 5
```

Comment:
I'm still not sure whether this is the best approach... I foresee issues with pylint contributors not knowing that this is an incorrect value..

Reply:
Oh great, forgot about pypy. In pypy the lines are correct, even with python 3.7. How should we handle that?

Reply:
This is how we've handled that in pydoctor: 

```python
_string_lineno_is_end = sys.version_info < (3,8) \
                    and platform.python_implementation() != 'PyPy'
"""True iff the 'lineno' attribute of an AST string node points to the last
line in the string, rather than the first line.
"""

def _set_docstring(self, ob: _model.ApiObject, node) -> None:
     docstring_lineno = node.lineno
        
     if _string_lineno_is_end:
            # In older CPython versions, the AST only tells us the end line
            # number and we must approximate the start line number.
            # This approximation is correct if the docstring does not contain
            # explicit newlines ('\n') or joined lines ('\' at end of line).
            docstring_lineno -= doc.count('\n')
```

Hope it helps

Reply:
That's really interesting actually, I didn't think about this approach. I guess we could do something as well?

I would think we could then reflect that `lineno` is an approximate in the docstring of the attribute and use `end_lineno` in `pylint` for the `bad-docstring-quotes` check. 

@kasium Would you be willing to explore if counting line breaks is a feasible solution for us?

Reply:
@DanielNoord I'm sorry, but I'm currently overloaded with other tasks. I might can take a look in the 1/2 weeks, if this is fine.

Reply:
That's more than fine! We're all volunteers here, we're grateful for every minute you put into this project! 😄

Reply:
Hey, sorry for taking so long. So, the solution works but has also some limitions:
```python

@mock.patch('mock.'
            'foo')
class Foo:
    """
    Blah
    """
```
--> will still return the wrong start line number. If we have no 100% solution I would rather leave it and wait for the deprecation for python 3.7. What do you think?

Reply:
@DanielNoord With the last changes, we now have access to the source code inside `TreeRebuilder`. Haven't tested it yet, but it should be possible to use tokenize to extract the correct positions.